src/Functions/FunctionHelpers.cpp

```diff
@@ -157,36 +156,14 @@ void validateFunctionArgumentTypes(const IFunction & func,
 {
     if (arguments.size() < mandatory_args.size() || arguments.size() > mandatory_args.size() + optional_args.size())
     {
-        auto join_argument_types = [](const auto & args, const String sep = ", ")
-        {
-            String result;
-            for (const auto & a : args)
-            {
-                using A = std::decay_t<decltype(a)>;
-                if constexpr (std::is_same_v<A, FunctionArgumentDescriptor>)
-                {
-                    if (a.argument_name)
-                        result += "'" + std::string(a.argument_name) + "' : ";
-                    if (a.expected_type_description)
-                        result += a.expected_type_description;
-                }
-                else if constexpr (std::is_same_v<A, ColumnWithTypeAndName>)
-                    result += a.type->getName();
-
-                result += sep;
-            }
-
-            if (!args.empty())
-                result.erase(result.end() - sep.length(), result.end());
-
-            return result;
-        };
-
         throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
-            "Incorrect number of arguments for function {} provided {}{}, expected {}{} ({}{})",
-            func.getName(), arguments.size(), (!arguments.empty() ? " (" + join_argument_types(arguments) + ")" : String{}),
-            mandatory_args.size(), (!optional_args.empty() ? " to " + std::to_string(mandatory_args.size() + optional_args.size()) : ""),
-            join_argument_types(mandatory_args), (!optional_args.empty() ? ", [" + join_argument_types(optional_args) + "]" : ""));
+            "Incorrect number of arguments for function '{}'. Provided {}, expected {}",
+            func.getName(),
+            std::to_string(arguments.size()) + " argument" + String(arguments.size() != 1 ? "s" : ""),
```

Comment:
```suggestion
            fmt::format("{} argument{}", arguments.size(), arguments.size() != 1 "s" : ""),
```

Reply:
you can also extract this into a function which returns suffix based on the size so you don't repeat the same check always

Reply:
Done and done.